src/main/java/com/example/demo/dao/GradeLocalRepository.java

```diff
@@ -0,0 +1,73 @@
+package com.example.demo.dao;
+
+import com.example.demo.dao.entity.Grade;
+import org.springframework.stereotype.Repository;
+
+import java.util.*;
+
+@Repository
+public class GradeLocalRepository implements GradeRepository {
+    private Map<Long, List<Grade>> gradeRegister;
+
+    public GradeLocalRepository() {
+        this.gradeRegister = new HashMap<>();
+    }
+
+    @Override
+    public List<Grade> getAllGrades() {
+        List<Grade> allGradesList = new ArrayList<>();
+        gradeRegister.values().stream()
+                .forEach(list -> allGradesList.addAll(list));
+        return allGradesList;
+    }
+
+    @Override
+    public List<Grade> getStudentGrades(long studentId) {
+        List<Grade> studentGradeList;
+        studentGradeList = gradeRegister.getOrDefault(studentId, List.of());
+        return studentGradeList;
+    }
+
+    @Override
+    public void addGrade(Grade grade) {
```

Comment:
this mozna wyrzucic, wiadomo ze chodzi o ten konkretny obiekt
Tutaj pojawia sie zbędna logika o ktorej wspomnialem wyzej, ocena szuka listy w którą moze wskoczyć, wymusza to bardzo duzo operacji, tracimy zaletę HashMapy ktora ma byc szybka i wydajna, dodatkowo w tym momencie masz mutowalne obiekty domenowe wiec bez problemu mozesz dodac ocene do listy która znajdziesz po kluczu. Natomiast jak wprowadzisz DTOsy metoda add(grade) sie nie powiedzie, ale to na spokojnie :)

Reply:
Inaczej wygląda to w nowszej wersji aplikacji, do problemu wrócę robiąc DTO :P